django/db/models/lookups.py

```diff
@@ -119,6 +119,10 @@ def process_rhs(self, compiler, connection):
             value = value.resolve_expression(compiler.query)
         if hasattr(value, "as_sql"):
             sql, params = compiler.compile(value)
+            if isinstance(value, ColPairs):
+                raise ValueError(
+                    "CompositePrimaryKey cannot be used as a right-hand side."
+                )
```

Comment:
If this is the right approach (for now), maybe we want to document this.

Reply:
I know right-hand side is referred in the docs to when making your own lookups, but I think maybe it's not that well known. Perhaps `CompositePrimaryKey cannot be used as a lookup value.` :thinking: